packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -81,9 +81,6 @@ abstract class AssetBundle {
   /// isolate to avoid jank on the main thread.
   Future<String> loadString(String key, { bool cache = true }) async {
     final ByteData data = await load(key);
-    if (data == null) {
```

Comment:
Don't we want to keep this here?

Reply:
You see, the `data` here is not nullable, so it should be a unnecessary check IMO.

Reply:
And I'm also wondering why the analyzer doesn't report about it.

Reply:
Ahhh interesting, ok. Maybe because of unsound null safety?

Reply:
I'll take another look to see if there are any issues with the linter rule.